src/methods/join.js

```diff
@@ -445,10 +458,29 @@ export class JoinImplementation {
     let empty_effects = construct_empty_effects(realm);
     let v = realm.intrinsics.empty;
     if (c.consequent instanceof ReturnCompletion) {
-      let pnc = new PossiblyNormalCompletion(v, c.joinCondition, v, empty_effects, c.alternate, c.alternateEffects);
+      let pathCondtion = [(AbstractValue.createFromUnaryOp(realm, "!", c.joinCondition): any)];
```

Comment:
Open to suggestions on this one. I don't understand why `AbstractValue.createFromUnaryOp` returns a  `Value` instead of an `AbstractValue`.

Reply:
Every time you create an abstract value it is simplified in the current path context. This can result in a concrete value.

Reply:
I see. Makes sense. Thanks